src/main.cpp

```diff
@@ -4638,6 +4635,31 @@ bool CheckWork(const CBlock block, CBlockIndex* const pindexPrev)
     return true;
 }
 
+bool CheckBlockTime(const CBlockHeader& block, CValidationState& state, CBlockIndex* const pindexPrev)
+{
+    // Not enforced on RegTest
+    if (Params().NetworkID() == CBaseChainParams::REGTEST)
+        return true;
+
+    const int64_t blockTime = block.GetBlockTime();
+    const int blockHeight = pindexPrev->nHeight + 1;
+
+    // Check blocktime against future drift (WANT: blk_time <= Now + MaxDrift)
+    if (blockTime > pindexPrev->MaxFutureBlockTime())
+        return state.Invalid(error("%s : block timestamp too far in the future", __func__), REJECT_INVALID, "time-too-new");
```

Comment:
Is there a special reason why this test is not `state.DoS`ed?

Reply:
because of the short time window and the possibility of end up banning "good" nodes that have their internal clock few seconds different.
It's a "give the benefit of the doubt", every node will have a time offset notification that will alert them if their clock is networkely not acceptable.
 
Saying that, sounds good to add a limit threshold for the received time-wise invalid blocks per node and then ban them there (could be done in a subsequent PR).

Reply:
@Mrs-X that check's failure is kept `state.Invalid()` as it was before (and as it is upstream) because it's dependent on the current adjusted time of the node.
It's the other one that changed: now it's `state.DoS(100)` because it's not dependent on the median time past anymore, but on the previous block time (which is a constant).